### PR TITLE
Tbtc Maintainer: Added the maintainer command

### PR DESCRIPTION
**Pull request by tomaszslabon**
This PR adds a command for launching the maintainer for Bitcoin difficulty.
The relay maintainer can be launched using the command:
```
KEEP_ETHEREUM_PASSWORD="password" ./keep-client maintainer  --bitcoinDifficulty  --config /Users/tslabon/projekt/keep-core/configs/maintainer.toml --developer
```

In the future we can add other maintainer tasks and launch the maintainer, e.g.:
```
KEEP_ETHEREUM_PASSWORD="password" ./keep-client maintainer  --bitcoinDifficulty --spv  --config /Users/tslabon/projekt/keep-core/configs/maintainer.toml --developer
```

Maintainer uses 3 groups of configuration flags:
- `ethereum` (url, keyFile, etc.)
- `electrum` (url) 
- `maintainer` (bitcoinDifficulty, spv, etc.)

All the options can be provided on the command line or using a configuration file or a mix of them.
Here is an example configuration toml file:
```
[ethereum]
  URL = "ws://127.0.0.1:8546"
  KeyFile = "/path/to/ethereum/data/keystore/UTC--2022-10-03T10-45-23.667111000Z--7ca4c3c109851221b6560c7d4f3917db6d056977"

[bitcoin.electrum]
  URL = "https://blockstream.info/api/"

[tbtc.maintainer]
 BitcoinDifficulty = true
```

Tested locally.

### CODE REVIEW

**Comment by lukasz-zimnoch** [comment]
I'll open a PR with the bitcoin groundwork soon. Please hold off with the merge of this PR until so.

**Comment by lukasz-zimnoch** [comment]
The PR with the Bitcoin toys is ready: https://github.com/keep-network/keep-core/pull/3384

File: cmd/relay.go
Comment (lukasz-zimnoch) [inline cmd/relay.go:15]: I think we must think about this command a little bit wider. The package `pkg/maintainer` will hold not only the relay maintainer but also the SPV maintainer code, wallet lifecycle maintainer code, and so on. That said, the `./keep-client relay` command is probably not a good fit here and we have basically two options to run that stuff:

Option 1: A bulk run of all maintainers with a single command. In this option, we would have a `./keep-client maintainer` command that will trigger all the maintainers in a single shot. This is more convenient but less flexible.

Option 2: A granular approach where each maintainer is triggered separately. In this option, we would have `./keep-client maintainer relay`, `./keep-client maintainer spv`, and so on. This is the exact opposite of the first option in terms of pros and cons.

I think we need to agree on a solution here @pdyraga @nkuba

File: pkg/maintainer/relay.go
Comment (lukasz-zimnoch) [inline pkg/maintainer/relay.go:13]: I'd move the config-related stuff to a separate file `pkg/maintainer/config.go` and cover the functions with unit tests.

File: pkg/maintainer/relay.go
Comment (lukasz-zimnoch) [inline pkg/maintainer/relay.go:84]: I think this is wrong. Ideally, the `pkg/maintainer` should be independent of `pkg/tbtc`. That means we should declare our own `maintainer.RelayChain` interface with expectations regarding the chain required by the relay maintainer and implement it within `pkg/chain/ethereum`.

**Review by lukasz-zimnoch — CHANGES_REQUESTED** [review]

File: pkg/maintainer/relay.go
Comment (tomaszslabon) [reply pkg/maintainer/relay.go]: Done in 1a02c7248bce4395be3830bc8b5bc4646f63f12a.

File: cmd/relay.go
Comment (nkuba) [reply cmd/relay.go:15]: I like the idea of introducing the `maintainer` command. To decide between Option 1 and Option 2 we need to consider how much of the config and command's flags/args will be common for all the types of maintiners.

Maybe we could use something like:
`./keep-client maintainer --relay`
`./keep-client maintainer --spv`
`./keep-client maintainer --relay --spv`
`./keep-client maintainer` <- this one will run all the maintainers

Comment (pdyraga) [reply cmd/relay.go:15]: I like @nkuba's proposition.

Also, I think we should mark `maintainer` feature as experimental in help. This will be in-development feature and we'll release new versions in the meantime.

Comment (tomaszslabon) [reply cmd/relay.go]: Done. I changed the way the maintainer tasks are started.
Now it's started like that:
`KEEP_ETHEREUM_PASSWORD="password" ./keep-client maintainer  --relay  --config /path/to/config/maintainer.toml`

The `KEEP_ETHEREUM_PASSWORD` environment variable is needed because we use ethereum configuration to start the maintainer.

In the future more maintainers could be started by providing more flags:
`KEEP_ETHEREUM_PASSWORD="password" ./keep-client maintainer  --relay --spv  --config /path/to/config/maintainer.toml`

File: pkg/maintainer/relay.go
Comment (tomaszslabon) [reply pkg/maintainer/relay.go]: I moved the config to `pkg/maintainer/config.go`, but I didn't add any unit tests, since I simplified the config and now it's a simple structure without any methods.

File: pkg/bitcoin/chain.go
Comment (lukasz-zimnoch) [inline pkg/bitcoin/chain.go:8]: This is the wrong place for that. This config is implementation-specific and should be exposed by `bitcoin.Client` implementations.

Comment (tomaszslabon) [reply pkg/bitcoin/chain.go:8]: I wanted to put this `Config` struct, so that we have a clean `bitcoin.Config` field in the general client configuration that any possible command of `keep-client` could reuse:
```
// Config is the top level config structure.
 type Config struct {
 	Ethereum   commonEthereum.Config
 	Bitcoin    bitcoin.Config
 	LibP2P     libp2p.Config `mapstructure:"network"`
 	Storage    storage.Config
 	ClientInfo clientinfo.Config
 	Maintainer maintainer.Config
 	Tbtc       tbtc.Config
 }
 ```
 
 My reasoning was that `URL`, `username` and`password` fields inside bitcoin.Config` are pretty generic and will fit any bitcoin implementation.
 
Should I move this config into the electrum package, so that we have 
```Electrum   electrum.Config```
inside the client config?

Comment (lukasz-zimnoch) [reply pkg/bitcoin/chain.go:8]: The `pkg/bitcoin` should only contain types from the Bitcoin protocol specification (see package documentation). There is no such type as `Config` in the Bitcoin spec. This is a pure implementation-specific concept and it should live in the implementation package. Also, different implementations could have different configs, one of them can use a username and password but another one can use an authentication token. This is another argument that we should keep the config on the implementation level. Last but not least, note that the `Ethereum commonEthereum.Config` is also implementation-specific, we don't declare it on the `pkg/chain` level. So as `LibP2P libp2p.Config` which is libp2p-specific and is not declared on `pkg/net` level. That being said, I think we should declare `Electrum electrum.Config` just as you proposed.

File: cmd/relay.go
Comment (lukasz-zimnoch) [reply cmd/relay.go]: Can we consider this resolved? I recall @tomaszslabon mentioned something about problems with the presented approach. What's the status here?

File: pkg/maintainer/relaychain.go
Comment (lukasz-zimnoch) [inline pkg/maintainer/relaychain.go:9]: What's the point of putting `maintainer.RelayChain` into a separate file? This won't be a huge interface so I'd keep it at the point of use (`relay.go`) to avoid jumping between files. Also, this is a Go-like way of organizing things, we don't need to create a separate file for each public type.

File: pkg/maintainer/maintainer.go
Comment (lukasz-zimnoch) [inline pkg/maintainer/maintainer.go:18]: I think we should use dependency injection here and make `Initialize` accept `bitcoin.Chain` and `RelayChain` as parameters, just like we do in `beacon.Initialize` and `tbtc.Initialize`. Ideally, package `pkg/maintainer` should depend on abstractions of both chains and should be not responsible for instantiating concrete implementations.

File: pkg/maintainer/relay.go
Comment (lukasz-zimnoch) [inline pkg/maintainer/relay.go:32]: I think we can make it simpler:
```suggestion
	chain RelayChain
```

File: pkg/bitcoin/chain.go
Comment (tomaszslabon) [reply pkg/bitcoin/chain.go]: Replaced `Bitcoin` configuration with `Electrum` in 2eb93eaefef4c21cbbcb13e47a1f1c465fe81576.
Once https://github.com/keep-network/keep-core/pull/3395 is merged to main, we will replace `Electrum` with `Electrs`. For now, let's stick to `Electrum`, as that's the name we have in the base branch of my PR.

File: pkg/maintainer/relaychain.go
Comment (tomaszslabon) [reply pkg/maintainer/relaychain.go]: Moved it back to `relay.go` in 7e975d796e0dd9de239fc80b0489538061d221ec.
I think the interface will have about 6 methods (there will be more methods than just `Retarget()`, e.g.  `ProofLength()`, `CurrentEpoch()`, `Ready()`, etc. ).

File: cmd/relay.go
Comment (tomaszslabon) [reply cmd/relay.go]: @lukasz-zimnoch 
My concern was whether launching multiple maintainers with one command is really beneficial.
I left a TODO to consider whether or not we should do that.

But I don't think this issue is a blocker for this PR, as we have just one maintainer now.
If we decide we don't want multiple maintainers run within one process, we will have disallow providing keep-client with multiple maintainer flags.

Comment (lukasz-zimnoch) [reply cmd/relay.go]: Ok, works for me! Thanks for clarification!

File: pkg/bitcoin/chain.go
Comment (lukasz-zimnoch) [reply pkg/bitcoin/chain.go]: > For now, let's stick to Electrum, as that's the name we have in the base branch of my PR.

We can change it even now. The `pkg/bitcoin/electrum` was just supposed to demonstrate the intended package structure. Maybe we can coordinate with @nkuba and change the config accordingly in this PR to have that covered?

Comment (tomaszslabon) [reply pkg/bitcoin/chain.go]: I added the same Config as @nkuba in #3395 in 9152aed1f0c75575d09f60d66bb71ffe35a558d7.
Updated all the code changed by this PR accordingly.
@nkuba should now have conflicts with this PR.

**Review by lukasz-zimnoch — DISMISSED** [review]
LGTM! Letting the config-related things in @nkuba hands!

File: cmd/maintainer.go
Comment (pdyraga) [inline cmd/maintainer.go:16]:
```suggestion
	Short: `Starts tBTC maintainer`,
```

File: cmd/maintainer.go
Comment (pdyraga) [inline cmd/maintainer.go:17]:
```suggestion
	Long:  `The maintainer command starts tBTC maintainer`,
```

File: pkg/maintainer/relay.go
Comment (pdyraga) [inline pkg/maintainer/relay.go:45]:
```suggestion
	logger.Info("starting Bitcoin difficulty maintainer")
```

File: pkg/maintainer/relay.go
Comment (pdyraga) [inline pkg/maintainer/relay.go:48]:
```suggestion
		logger.Info("stopping Bitcoin difficulty maintainer")
```

File: test/config.toml
Comment (pdyraga) [inline test/config.toml:13]: Given the recent discussion about supporting ElectrumX RPC API, I am not sure if this should be named `[Electrs]` or `[Electrum]`. @nkuba thoughts?

File: test/config.toml
Comment (pdyraga) [inline test/config.toml:36]: I would start calling it `BitcoinDifficulty` instead of the `Relay`. I think the `Relay` is pretty cryptic. We've been also using "threshold relay" as an alternative name for the random beacon.

File: pkg/maintainer/maintainer.go
Comment (pdyraga) [inline pkg/maintainer/maintainer.go:22]: That's the same problem as with the client handling multiple wallets 🤷🏻 

I think we should allow launching all types of maintainers at once. Even if the service crashed, the service should get restarted.

File: cmd/maintainer.go
Comment (tomaszslabon) [reply cmd/maintainer.go]: Done in 9fe493876ecf200cc755932defeb2498e0026d16.

File: cmd/maintainer.go
Comment (tomaszslabon) [reply cmd/maintainer.go]: Done in 9fe493876ecf200cc755932defeb2498e0026d16.

File: pkg/maintainer/relay.go
Comment (tomaszslabon) [reply pkg/maintainer/relay.go]: Done in 9fe493876ecf200cc755932defeb2498e0026d16.

File: pkg/maintainer/relay.go
Comment (tomaszslabon) [reply pkg/maintainer/relay.go]: Done in 9fe493876ecf200cc755932defeb2498e0026d16.

File: pkg/maintainer/maintainer.go
Comment (tomaszslabon) [reply pkg/maintainer/maintainer.go]: OK.
I updated the TODO in 43106ec2d46347892753e797b45acc83a0f9eb1d.

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml:36]: @pdyraga 
Should I also change the option for starting from `./keep-client maintainer --relay` to `./keep-client maintainer --bitcoinDifficulty`?

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml:13]: @nkuba @pdyraga 
Alternatively, we could keep the name as `electrs` and treat it as kind of a placeholder and update it once [Kuba's PR](https://github.com/keep-network/keep-core/pull/3406) is merged and we know what config is used there and which fields are required to be configured.

Comment (nkuba) [reply test/config.toml:13]: We could introduce a parent `bitcoin` section, so:
```
[bitcoin.electrum]
URL=.....
```

If we decide to keep multiple APIs supported we could add them in te future as in the example below:

```
[bitcoin]
[electrum]
URL=.....
[electrs]
URL=....
```

Comment (pdyraga) [reply test/config.toml:13]: I think `[bitcoin]` section makes a lot of sense, especially since we have `[ethereum]`. I just wonder if we couldn't keep it simple like:
```
[bitcoin]
  ElectrumURL = ""
```

ElectrumURL does not mean it has to be ElectrumX or Electrs. That's just an Electrum API-compatible node.

Comment (pdyraga) [reply test/config.toml:13]: OK, I've just realized there are more options for electrum than just URL. Then I think `[bitcoin.electrum]` is the best one for now.

File: test/config.toml
Comment (pdyraga) [inline test/config.toml:35]: What do you think about renaming it to `[tbtc.maintainer]`? The reasoning is that it's not a generic maintainer but a maintainer specific for tBTC. That would play nicely with the idea of `[bitcoin.electrum]` I think.

File: cmd/cmd.go
Comment (pdyraga) [inline cmd/cmd.go:38]:
```suggestion
		TbtcMaintainerCommand,
```

File: cmd/flags.go
Comment (pdyraga) [inline cmd/flags.go:49]:
```suggestion
		case config.TbtcMaintainer:
```

File: cmd/flags.go
Comment (pdyraga) [inline cmd/flags.go:39]: If we do in the config `[bitcoin.electrum]`, then this one should probably become:

```suggestion
		case config.BictoinElectrum:
```

File: cmd/flags.go
Comment (pdyraga) [inline cmd/flags.go:159]: Similarly to `config.BitcoinElectrum`, if we do `[bitcoin.electrum]` in the config, then this one should probably be:
```suggestion
func initBitcoinElectrumFlags(cmd *cobra.Command, cfg *config.Config) {
```

File: cmd/flags.go
Comment (tomaszslabon) [reply cmd/flags.go]: Done in b0285cf27f51896294dff419b3dec8cf416e7b49.

File: cmd/maintainer.go
Comment (tomaszslabon) [reply cmd/maintainer.go]: Done in 3fca39275fcfb17c63d9fdf8553f8d369d79a629.

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml]: Done in https://github.com/keep-network/keep-core/blob/b26f5297258e43a7f57047a0157e5f3b111adab5/pkg/bitcoin/config/config.go#L8.

Unfortunately, there is some complication: in order to have `bitcoin.electrum` configuration, I had to define [this Config struct](https://github.com/keep-network/keep-core/blob/b26f5297258e43a7f57047a0157e5f3b111adab5/pkg/bitcoin/config/config.go#L8) somwhere.

I couldn't do this in the `bitcoin` package because of circular dependency problem (`electrum` imports `bitcoin` and `bitcoin` would import `electrum`). It would also mean that the specific bitcoin implementation (electrum) leaks into its parent package.

Therefore I decided to create a separate package `pkg/bitcoin/config` that would contain the configs for all the possible bitcoin implementations (electrum, electrs, etc).

I don't like this idea too much. Are you ok with this or should I do sth else?

File: cmd/flags.go
Comment (tomaszslabon) [reply cmd/flags.go]: Done in b26f5297258e43a7f57047a0157e5f3b111adab5.

File: cmd/flags.go
Comment (tomaszslabon) [reply cmd/flags.go]: Done in b26f5297258e43a7f57047a0157e5f3b111adab5.

File: test/config.toml
Comment (lukasz-zimnoch) [reply test/config.toml]: Why not just in `pkg/config`? Package `pkg/bitcoin` does not have to contain it.

Comment (tomaszslabon) [reply test/config.toml]: @lukasz-zimnoch 
Yes, I also thought about it.
Done in 96b0e94e3ce9f1faca845bc3e2a3ee5a26a100b5.

BTW, should it be `keep-core/pkg/config` or simply `keep-core/config`? I chose the first option, but I had to create a new directory for `pkg/config`.

File: pkg/maintainer/bitcoindifficulty.go
Comment (tomaszslabon) [reply pkg/maintainer/bitcoindifficulty.go]: Done in 29ad4a69229b533c79cf9f19bc80f2e872bbf31a.

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml]: Done in ed67c93fbe4f57b8aaa434aca80e0100f535f31d.

File: test/config.toml
Comment (pdyraga) [reply test/config.toml]: `pkg/bitcoin` defines Bitcoin chain abstractions and `pkg/bitcoin/electrum` uses these abstractions. I think this dependency makes sense and we should leave it as-is.

On the other hand, all other configuration structures lie in respective packages now: `libp2p`, `storage`, `clientinfo`, `ethereum`... `pkg/config` with `BitcoinConfig` is the only exception which I am not a huge fan of. 

Wouldn't it be better to move `BitcoinConfig` to `pkg/bitcoin` and embed electrum configuration in this file? It is still not ideal but better in my opinion:

```
type BitcoinConfig struct {
   Electrum ElectrumConfig 
}

type ElectrumConfig struct {
	URL string
}
```

Alternatively, introduce `pkg/bitcoin/config` package. Given we have just one configuration option for electrum now, I'd prefer the first solution as the simplest one.

File: pkg/tbtc/maintainer/bitcoin_difficulty.go
Comment (pdyraga) [inline pkg/tbtc/maintainer/bitcoin_difficulty.go:19]: We could also extract it to `chain.go` and name it just `Chain`. This would be consistent with what we do in `pkg/sortition` and I believe that was our desired organization of packages interacting with chain for v2 client.

**Comment by pdyraga** [comment]
I tried to test it starting the maintainer with `KEEP_ETHEREUM_PASSWORD="password" ./keep-client --config configs/config.1.toml maintainer` but nothing is printed to the console and it looks like if the process hanged. What am I doing wrong?

**Comment by tomaszslabon** [comment]
> I tried to test it starting the maintainer with `KEEP_ETHEREUM_PASSWORD="password" ./keep-client --config configs/config.1.toml maintainer` but nothing is printed to the console and it looks like if the process hanged. What am I doing wrong?

You need to provide the `--bitcoinDifficulty` flag: 
```
KEEP_ETHEREUM_PASSWORD="password" ./keep-client maintainer  --bitcoinDifficulty  --config /Users/tslabon/projekt/keep-core/configs/maintainer.toml 
```

File: test/config.toml
Comment (pdyraga) [reply test/config.toml]: I looked at `pkg/bitcoin` one more time and you are right @lukasz-zimnoch. This package is just about Bitcoin types, as the referenced documentation explains. From this perspective, we are not configuring anything in `pkg/bitcoin` but we are configuring Electrum.

From the config file's perspective though, I think it makes sense to have the `bitcoin` section, especially that we have `ethereum` section with an URL there and we do not care if it is Geth, Nethermind, Erigon, or others.

What about having`[bitcoin.electrum]` in the config file, just like we do now but having the config structure in `pkg/bitcoin/electrum`?

**Comment by pdyraga** [comment]
I was under the impression that we start all maintainers by default?

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml]: > What about having[bitcoin.electrum] in the config file, just like we do now but having the config structure in pkg/bitcoin/electrum?

But what if we ever want to add another typy of Bitcoin implementation (e.g. `electrs`) and use them both, so that we have:
```
[bitcoin]
  [electrum]
    URL = ...
  [electrs]
    URL = ... 
```

We would be defining the Bitcoin structure with `electrum` and `electrs` in electrum?

**Comment by tomaszslabon** [comment]
> I was under the impression that we start all maintainers by default?
I [left a TODO](https://github.com/keep-network/keep-core/blob/29ad4a69229b533c79cf9f19bc80f2e872bbf31a/pkg/tbtc/maintainer/maintainer.go#L21) for that once we have more than one maintainer.

File: test/config.toml
Comment (lukasz-zimnoch) [reply test/config.toml]: Yeah, this is the best option IMO. Having `bitcoin` section is a pure config thing and it should be declared there.

File: cmd/flags.go
Comment (tomaszslabon) [reply cmd/flags.go:163]: There are already 3 additional options in that [config](https://github.com/keep-network/keep-core/blob/d69e971fc734031111699f95eccc3bbbafe4345f/pkg/bitcoin/electrum/config.go#L16).

File: cmd/flags_test.go
Comment (tomaszslabon) [reply cmd/flags_test.go:329]: This test uses [config_flags.toml](https://github.com/keep-network/keep-core/blob/29ad4a69229b533c79cf9f19bc80f2e872bbf31a/test/config_flags.toml#L10) and the value for `bitcoin.electrum.url` is different there.

File: cmd/tbtc_maintainer.go
Comment (tomaszslabon) [reply cmd/tbtc_maintainer.go]: Done in 7685dacdbce95115282ba34cc661eab347ffcea2.

File: config/category.go
Comment (tomaszslabon) [reply config/category.go:26]: Done in f41b9f9ab7152ffe20f2bb779ec6bf61d0b0a6ec.

File: test/config.toml
Comment (tomaszslabon) [reply test/config.toml]: Moved the `BitcoinConfig` struct to the electrum package in 232b141cac2dc47b6aff6b3e24c924a85aeb157a.
Hope this is what you wanted.

File: pkg/tbtc/maintainer/bitcoin_difficulty.go
Comment (tomaszslabon) [reply pkg/tbtc/maintainer/bitcoin_difficulty.go]: Done in 13487d5617e34cb6ae4a17b0001baa452fd57229.
I moved the `BitcoinDifficultyChain` into `chain.go`. 
But I didn't name that structure `Chain`, because we will have more than one maintainer interface (e.g.`SpvChain`).
Is that ok?

File: pkg/bitcoin/electrum/config.go
Comment (pdyraga) [inline pkg/bitcoin/electrum/config.go:4]: What about naming it just `Config`? We would have `electrum.Config` living in `pkg/bitcoin/electrum` referenced from `[bitcoin.electrum]` section. I think it all makes sense.

File: test/config.toml
Comment (pdyraga) [reply test/config.toml]: Yep!

File: cmd/flags.go
Comment (pdyraga) [reply cmd/flags.go:163]: I see... another PR with potentially conflicting changes. Cool, let's leave it this way!

File: pkg/bitcoin/electrum/config.go
Comment (tomaszslabon) [reply pkg/bitcoin/electrum/config.go:4]: There is already such a [structure](https://github.com/keep-network/keep-core/blob/c87ccffd4c948c25b629641e022c91451826225a/pkg/bitcoin/electrum/config.go#L10)
 in the electrum package and it contains config for electrum (it was added by Kuba and I copied it). It's the structure used by BitcoinConfig.

Comment (pdyraga) [reply pkg/bitcoin/electrum/config.go:4]: Right, I think we should have just one config (cc @nkuba) structure named `Config`:
```
package electrum

// Config defines the configuration for the Electrum Bitcoin client.
type Config struct {
  // URL to the Electrum server in format: `hostname:port`.
  URL string
}
```

Based on [this](https://github.com/keep-network/keep-core/pull/3374#discussion_r1024157900) conversation, we agreed that we are *not* configuring Bitcoin but we are configuring Electrum Bitcoin client. The configuration is in `[bitcon.electrum]` in the config and the structure is in `pkg/bitcoin/electrum/config.go`.

Comment (tomaszslabon) [reply pkg/bitcoin/electrum/config.go:4]: If we want to use a nested structure in the config file:
```
[bitcoin.electrum]
URL = "url.to.electrum:1111"
```
then the corresponding structure in Golang should also be nested:
```
type BitcoinConfig struct {
 Electrum Config
}

type Config struct {
 	URL string
}
```.

If we don't use a nested structure, but just a simple non-nested `Config` structure, then the `Viper` package that we use for mapping config file to the Golang structure will be unable to read the configuration properly.

**Review by pdyraga — APPROVED** [review]

File: pkg/bitcoin/electrum/config.go
Comment (nkuba) [reply pkg/bitcoin/electrum/config.go:4]: I proposed an update in https://github.com/keep-network/keep-core/pull/3406/commits/40ff528ac1797d6ce3cf2a449ef745e10480b538

**Pull request merged**
